### ISSUE TITLE
Fix pythonjsonlogger deprecation warning

### ISSUE THREAD

**Issue by MarcusJellinghaus** [MarcusJellinghaus/mcp-code-checker#70]
## Description

  The `log_utils.py` file uses a deprecated import that triggers a `DeprecationWarning` in pytest output:

  DeprecationWarning: pythonjsonlogger.jsonlogger has been moved to pythonjsonlogger.json

  ## Current Code

  ```python
  from pythonjsonlogger import jsonlogger

  Fix

  Update the import to use the new module path:

  from pythonjsonlogger.json import JsonFormatter

  And update any references from jsonlogger.JsonFormatter to JsonFormatter.

  Impact

  This warning appears 4 times in CI test output, adding noise to test results.
